sql/moz-fx-data-shared-prod/telemetry_derived/cohort_daily_statistics_v2/query.sql

```diff
@@ -0,0 +1,150 @@
+    app_version,
+    attribution_campaign,
+    attribution_content,
+    attribution_experiment,
+    attribution_medium,
+    attribution_source,
+    attribution_ua,
+    -- attribution_variation, -- can find in clients_daily_v6 with unpacking attribution field
+    partner_distribution_version,
+    partner_distributor,
+    partner_distributor_channel,
+    partner_id,
+    city,
+    country,
+    -- device_model, -- important for mobile, usually null, may not need
+    apple_model_id, -- from clients_first_seen_v2
+    distribution_id,
+    -- is_default_browser,  -- can find in clients_daily_v6
+    locale,
+    -- normalized_app_name,  -- strings that come from unified_metrics/filters out BrowserStack
+    app_name as normalized_app_name, -- from clients_first_seen_v2, as per Lucia's msg app_name = normalized_app_name
+    normalized_channel,
+    normalized_os,
+    normalized_os_version,
+    -- os_version_major,
+    -- os_version_minor,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(0)], "") AS INTEGER),
+      0
+    ) AS os_version_major,
+    COALESCE(
+      SAFE_CAST(NULLIF(SPLIT(normalized_os_version, ".")[SAFE_OFFSET(1)], "") AS INTEGER),
+      0
+    ) AS os_version_minor,
+  FROM
+    telemetry_derived.clients_first_seen_v2
+  WHERE
+    first_seen_date
+    BETWEEN DATE_SUB(@activity_date, INTERVAL 180 DAY)
+    AND DATE_SUB(@activity_date, INTERVAL 1 DAY)
+    -- (1) No need to get activity for the cohort created on activity_date - everyone will be retained
+    -- (2) Note this is a pretty big scan... Look here for problems
+),
+activity_cohort_match AS (
+  SELECT
+    cohorts_in_range.client_id AS client_id,
+    submission_date_activity.client_id AS active_client_id,
+    submission_date_activity.days_seen_bits as active_client_days_seen_bits,
+    mozfun.bits28.days_since_seen(submission_date_activity.days_seen_bits) as active_clients_days_since_seen,
+    cohorts_in_range.* EXCEPT (client_id)
+  FROM
+    cohorts_in_range
+  LEFT JOIN
+    submission_date_activity
+  USING
+    (client_id, activity_date)
+)
+SELECT
+  cohort_date,
+  activity_date,
+  -- activity_segment,
+  app_version,
+  attribution_campaign,
+  attribution_content,
+  attribution_experiment,
+  attribution_medium,
+  attribution_source,
+  attribution_ua,
+  -- attribution_variation,
+  partner_distribution_version,
+  partner_distributor,
+  partner_distributor_channel,
+  partner_id,
+  city,
+  country,
+  -- device_model,
+  apple_model_id, -- from clients_first_seen_v2
+  distribution_id,
+  -- is_default_browser,
+  locale,
+  normalized_app_name,
+  normalized_channel,
+  normalized_os,
+  normalized_os_version,
+  os_version_major,
+  os_version_minor,
+  COUNT(client_id) AS num_clients_in_cohort,
+  COUNTIF((active_client_id IS NOT NULL) AND (active_clients_days_since_seen = 0)) AS num_clients_active_on_day,
```

Comment:
We should probably document how people should use this table to calculate the 28 day and 7 day retention measures, since the queries aren't too bad but neither are they that obvious. For 28 day retention, I think it would be something like this, for individual cohorts:

```sql
SELECT
    cohort_date,
    SAFE_DIVIDE(DATE_DIFF(activity_date, cohort_date, DAY) + 1, 28) AS month,
    SAFE_DIVIDE(num_clients_active_atleastonce_in_last_28_days, num_clients_in_cohort) AS retention_rate
FROM cohort_daily_statistics
WHERE MOD(DATE_DIFF(activity_date, cohort_date, DAY) + 1, 28) = 0
ORDER BY cohort_date, activity_date
```
(I haven't tested that code exactly, but that should be roughly it.)

The idea is to measure at equally spaced 28 day intervals.

Reply:
Will test your code and give this a think. Update tomorrow

Reply:
We should put this in the lookml stage.

Reply:
sounds good, will put the info in at the lookml stage